### PR TITLE
Worker refactor

### PR DESCRIPTION
**Pull request by pvdvreede**
This refactor is pulling out the logic for each type of event into a Worker.

Im thinking the steps will be:

1. Create worker interface
1. Setup worker to be inject impure dependencies for testing
1. Orchestrate the create event with db and secret packages with tests
1. Orchestrate the delete event

This should keep feature parity and be merged and deployed, and then from there add more functionality.

### CODE REVIEW

**Comment by pvdvreede** [comment]
Along the way I am also trying to reduce the surface area of the repo for things like the old dataform stuff down to only the things we need for the db operator.

Less code == less complexity == less issues == faster development

File: main.go [hunk 1]
```diff
@@ -36,50 +41,42 @@ func main() {
...
-		log.Infof("running inside cluster")
+		glog.Infof("running inside cluster")
 		config, err = rest.InClusterConfig()
 	}
 
-	kubeClient, err := kubernetes.NewForConfig(config)
+	k8sClient, err := kubernetes.NewForConfig(config)
 	if err != nil {
-		log.Fatalf("error creating kubernetes client: %v", err)
+		glog.Fatalf("Error building k8s clientset: %s", err.Error())
```
Comment (weirdtales) [inline main.go]: `Fatalf()` will call `os.Exit()` after logging the error.

File: pkg/secret/db_secret.go [hunk 2]
```diff
@@ -29,6 +29,36 @@ type DBSecret struct {
...
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: d.Namespace,
+				Name:      d.Name,
+			},
+			StringData: d.Map(),
+		}
+		obj, err = d.client.Secrets(d.Namespace).Create(obj)
+
+		return err
+	} else if err != nil {
```
Comment (weirdtales) [inline pkg/secret/db_secret.go]: `else if` is not required if the first condition calls `return`:

```golang
if err != nil && errors.IsNotFound(err) {
    // ...
    return err
}
if err != nil {
    return err
}
```

this pattern improves "line of sight".

**Comment by pvdvreede** [comment]
Given the size of this PR I will merge it in and we will deal with the bugs and feature loss that is will no doubt have in it in separate PRs.

**Pull request merged**